File: framework/ui-elements/src/displays/components/Frame.js
import React from "react";
import { withStyles } from '@material-ui/core/styles';
import AbstractFrame from "../../../gen/displays/components/AbstractFrame";
import FrameNotifier from "../../../gen/displays/notifiers/FrameNotifier";
import FrameRequester from "../../../gen/displays/requesters/FrameRequester";
import DisplayFactory from "alexandria-ui-elements/src/displays/DisplayFactory";

const styles = theme => ({
	container : {
		width: "100%",
		height: "100%",
		border: "0",
		position: "relative",
	},
});

class Frame extends AbstractFrame {

	constructor(props) {
		super(props);
		this.notifier = new FrameNotifier(this);
		this.requester = new FrameRequester(this);
        this.state = {
            ...this.state,
            url : this.props.url
        };
	};

	render() {
		if (!this.state.visible) return (<React.Fragment/>);
		if (this.state.url == null) return (<React.Fragment/>);
		const { classes } = this.props;
		return (<div style={this.style()}><iframe src={this.state.url} className={classes.container}/></div>);
	};

    refresh = (url) => {
        this.setState({url});
    };

	style() {
		var result = super.style();
		if (result == null) result = {};
		if (this._widthDefined() && result.width == null) result.width = this.props.width;
		if (this._heightDefined() && result.height == null) result.height = this.props.height;
		return result;
	};
}

export default withStyles(styles, { withTheme: true })(Frame);
DisplayFactory.register("Frame", withStyles(styles, { withTheme: true })(Frame));

File: framework/ui/src/io/intino/alexandria/ui/displays/components/Frame.java
package io.intino.alexandria.ui.displays.components;

import io.intino.alexandria.core.Box;
import io.intino.alexandria.exceptions.*;
import io.intino.alexandria.*;
import io.intino.alexandria.schemas.*;
import io.intino.alexandria.UiFrameworkBox;
import io.intino.alexandria.ui.displays.components.AbstractFrame;
import io.intino.alexandria.ui.displays.notifiers.FrameNotifier;

public class Frame<DN extends FrameNotifier, B extends Box> extends AbstractFrame<B> {
    private String url;

    public Frame(B box) {
        super(box);
    }

    public String url() {
        return url;
    }

    public void url(String url) {
        _url(url).refresh();
    }

    protected Frame<DN, B> _url(String url) {
        this.url = url;
        return this;
    }

    @Override
    public void refresh() {
        super.refresh();
        notifier.refresh(url);
    }
}